first-service/src/main/java/com/assets/firstservice/service/FirstRepository.java

```diff
@@ -0,0 +1,49 @@
+package com.assets.firstservice.service;
+
+
+import com.assets.firstservice.domain.First;
+import java.util.List;
+import java.util.Optional;
+import lombok.NonNull;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Customized repository.
+ *
+ * Here are methods customized for using softDelete strategy
+ * @since 1.0.0-SNAPSHOT
+ * @author Rosyslav Balushchak
+ */
+@Repository
+public interface FirstRepository extends JpaRepository<First, Long> {
+
+  @NonNull
+  @Override
+  @Transactional(readOnly = true)
+  @Query(nativeQuery = true,
```

Comment:
why do you use native query here? why not jpql ?

Reply:
Because i though it has better performance, refactored.